clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1013,16 +1013,8 @@ private void maybeFailWithError() {
         if (!hasError()) {
             return;
         }
-        // for ProducerFencedException, do not wrap it as a KafkaException
```

Comment:
Did we ever get the rationale for why this decision was made? It seems to suggest here that the call stack will be incorrect so we would rather have an empty one?

Reply:
That's as much as I got from the comment, but I am not sure if it makes sense. Yes, producers are generally not fenced because of a specific operation, but because something happened on the server side, but it's still "correct" to include a call trace to the operation where the broker first returned a fencing error. At least, it doesn't seem worth complicating the exception handling with these inconsistencies to hide the call trace.